---
title: "Statistic analysis of metagenomics data"
output: html_document
date: "2022-10-31"
author: Xiangyu Li
---

### Set the working directory

In the R cloud directory section:

1.  Go to 'Metagenomics_Statistics_2022' directory
2.  Click 'More' icon and select "Set As Working Directory"

### Load the packages

```{r, message=F}
library(tidyr)
library(dplyr)
library(tibble)
library(phyloseq)
library(microbiome)
library(vegan)
```

Note:If there is any package which has been not installed, please go to the slides to find the codes.

### Load the read count data

```{r, message=F}
#-------------------------------
```

Note: The '.Rdata' is one type of R files. It is generated by R function 'save', for example: save(file = "phyloseq_object_read_2022.Rdata",phylo_obj,OTUdata,SampleData,TAXAData). It can save multiple variables in one Rdata at the same time.

### View the data

```{r}
#-------------------------------
```

```{r}
#-------------------------------
```

```{r}
#-------------------------------
```

### Section 1: Alpha diversity Analysis

Note: All alpha diversity estimators are calculated by the R function 'estimate_richness' in 'phyloseq' packages. In this section, we use the read count profiles.

#### 1.1 Estimate the 'Observed species'

```{r}
Adiv <- estimate_richness(phylo_obj, measures="Observed")
Adiv[1:5,]
```

#### 1.2 Task 1 and Questions

Task 1: Calculate the Chao 1 index and ACE index

```{r}
#-------------------------------
```

Questions:

1\. Why 'Observed' is equal to "Chao 1" in this data?

2\. Why ACE is equal to NAN?

Note: se.chao 1 and se.ACE represent the standard errors for the prediction of rare species with low-abundance in Chao 1 and ACE, respectively. Since there is no rare species in our practicing data, se.Chao 1 and se.ACE are 0 and NAN.

#### 1.3 Break

#### 1.4 Task 2: Estimate the 'Shannon','Simpson',and 'InvSimpson'

Tips: Function 'estimate_richness'

```{r}
#-------------------------------
```

#### 1.5 Break

### Section 2: Statistical Analysis

#### 2.1 Shapiro-Wilk normality Test

Note: R fuction 'shapiro.test'

Question: If I would like to test the the normal distribution of Shannon index, should I test for disease and control samples separately or together?

```{r}
#Get the location of samples
#----------------------
#Extract the alpha diversity values
#----------------------
#Shapiro-Wilk test
#----------------------
```

#### 2.2 Differential test for Shannon index

Note: R function 't.test' or 'wilcox.test'

```{r}
#----------------------
```

#### 2.3 Task 3

1.  Please make normality test and differential test for Simpson test.
2.  If there is significant difference of Simpson between groups, how could we know which group's Simpson index is higher?

```{r}
#-------------------------------
```

#### 2.4 Linear correlation

Note: R function 'cor.test'

Correlation between Shanno and Simpson

```{r}
#----------------------
```

#### 2.5 Task 4

Is there any alpha diversity estimator is associated with age?

Tips: For loop

```{r}
#-------------------------------
```

### Section 3. Beta Diversity Analysis

Note: For beta diversity calculation, we use the relative abundance profiles.

Remove all the current variables in the

```{r}
rm(list=ls())
```

Load the relative abundance profiles

```{r}
#-------------------------------
```

#### 3.1 Beta diversity

```{r}
#log10 transformation for the abundance data
phylo_obj <- microbiome::transform(phylo_obj,transform="log10")
#beta diversity based on Bray  distance 
Betdiv <-as.matrix(phyloseq::distance(phylo_obj, method="bray"))
Betdiv[1:5,1:5]
```

#### 3.2 PERMANOVA test for beta diversity

```{r}
result<-adonis2(Betdiv ~ Condition, data=SampleData, permutations=1000,method="bray")
result
```

P\<0.05, means the beta diversity is significantly different between the two groups. F is the statistic. Higher F value means bigger difference of diversity between groups. R2=9.98% means that 9.98% of the variance can be explained by the group difference.
